py/src/braintrust/logger.py

```diff
@@ -3353,20 +3346,55 @@ def __exit__(self, exc_type, exc_value, traceback) -> None:
         del exc_type, exc_value, traceback
 
 
-def render_message(render, message):
-    return {
-        **{k: v for (k, v) in message.as_dict().items() if v is not None},
-        "content": render(message.content)
-        if isinstance(message.content, str)
-        else [
-            {**c.as_dict(), "text": render(c.text)}
-            if c.type == "text"
-            else {**c.as_dict(), "image_url": {**c.image_url, "url": render(c.image_url.url)}}
-            if c.type == "image_url"
-            else c
-            for c in message.content
-        ],
-    }
+def render_message(render: Callable[[str], str], message: PromptMessage):
+    base = {k: v for (k, v) in message.as_dict().items() if v is not None}
+    # TODO: shouldn't load_prompt guarantee content is a PromptMessage?
```

Comment:
@ankrgyl would you expect this as well? I can track it down. I imagine load_prompt should be trying something akin to `PromptMessage(**serialized)`.